bigtable/beam/bulk-data-generator/src/main/java/BulkWrite.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.bigtable.admin.v2.Cluster;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.beam.CloudBigtableIO;
+import com.google.cloud.bigtable.beam.CloudBigtableTableConfiguration;
+import com.google.cloud.bigtable.grpc.BigtableClusterName;
+import com.google.cloud.bigtable.grpc.BigtableClusterUtilities;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.joda.time.Duration;
+
+public class BulkWrite {
+
+  static final double TB_PER_TABLE = .5;
+  static final long MB_PER_ROW = 1;
+  static final long ONE_MB = 1000 * 1000;
+  static final long ONE_GB = 1000 * ONE_MB;
+  static final long ONE_TB = 1000 * ONE_GB;
+  static final long MB_PER_SEC = 120;
+
+  static final String COLUMN_FAMILY = "cf";
+
+  public static void main(String[] args) {
+    BigtableOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigtableOptions.class);
+    Pipeline p = Pipeline.create(options);
+    System.out.println(options.getBigtableInstanceId());
+
+    BigtableTableAdminClient adminClient = null;
+    try {
```

Comment:
Should the IOException be allowed to bubble up to the next level?

Reply:
Or the pattern I see a lot is something like:

```
try (adminClient = BigtableTableAdminClient.create(adminSettings)) {
// more sample here
}
```

Reply:
I think this is still an issue in this code. 

We do something like this in the Vertex AI Java samples:
https://github.com/googleapis/java-aiplatform/blob/master/samples/snippets/src/main/java/aiplatform/CreateBatchPredictionJobTextClassificationSample.java#L61

Reply:
I can't do that because building the bigtable admin settings can throw an IOException

Reply:
The only error create throws is IOException, so I don't need the try if I have the throws IOException to Main